Lesson3

yaNimotsi opened the pull request:
PersonController method is work

Commit:
add PersonControlles and peson class to TimeSheetApi.
Add class library BuisnessLogic and PersonModel

Commit:
add StartLogic

Commit:
Controller method is work

Commit:
Update project name and namespaces

MeatBatyr commented on BuisnessLogic/Class1.cs:
```diff
@@ -0,0 +1,8 @@
+﻿using System;
+
+namespace BuisnessLogic
```
мусор

yaNimotsi replied:
Мусор удалил

MeatBatyr commented on PersonModel/PersonLogic.cs:
```diff
@@ -0,0 +1,75 @@
+{
+    public class PersonLogic
+    {
+        public Person AddNewPerson(Repository repository, Person newPerson)
```
Repository нужно внедрять через конструктор и использовать DI, а не параметром в каждый метод

yaNimotsi replied:
Внедрил IPersonRepository через DI

MeatBatyr commented on PersonRepository/Repository.cs:
```diff
@@ -0,0 +1,79 @@
+
+namespace TimeSheet.DB
+{
+    public class Repository
```
должен быть интерфейс для репозитория, и этот класс должен его реализовывать, и в итоге получится обертка над  private static List<Person> _personsList;

MeatBatyr replied:
возможно этот класс и не нужен, пожно просто переименовать PersonLogic и написать интерфейс

yaNimotsi replied:
Создал IRepository, IPersonRepository. Класс Repository сделал статическим.

MeatBatyr commented on PersonModel/PersonLogic.cs:
```diff
@@ -0,0 +1,75 @@
+
+namespace TimeSheet.BisnesLogic
+{
+    public class PersonLogic
```
по факту этот класс очень похожу на репозиторий

MeatBatyr commented on PersonModel/PersonLogic.cs:
```diff
@@ -0,0 +1,75 @@
+        {
+            if (skip >= repository.PersonList.Count)
+            {
+                return null;
```
лучше в таких случаях возвращать пустую коллекцию

yaNimotsi replied:
Если не сложно, то поясните на лекции почему

MeatBatyr commented on PersonModel/PersonLogic.cs:
```diff
@@ -0,0 +1,75 @@
+        {
+            if (!(repository.PersonList?.Count > 0)) return null;
+
+            var personToUpdate = GetPerson(repository, newPersonData.Id);
```
а если нет с таким айдишником записи?

yaNimotsi replied:
Исправил

MeatBatyr commented on PersonModel/PersonLogic.cs:
```diff
@@ -0,0 +1,75 @@
+        {
+            if (!(repository.PersonList?.Count > 0)) return false;
+
+            var personToDelete = GetPerson(repository, id);
```
а если нет с таким айдишником записи?

yaNimotsi replied:
Исправил

MeatBatyr commented on TimeSheet/Controllers/PersonController.cs:
```diff
@@ -0,0 +1,61 @@
+        // GET: PersonController
+        public IActionResult GetPersonById([FromRoute] int id)
+        {
+            return Ok(_personLogic.GetPerson(_repository, id));
```
а если не нашел, тот тоже 200 возвращаем?

yaNimotsi replied:
Исправил

Commit:
Update after revie.
DB layer - add IRepository, IPersonReposirory. Now class Repostitory is static.
BL layer - Update person logic
API layer - Update return parametter in PersonController
